<img src="http://imgur.com/1ZcRyrc.png" style="float: left; margin: 20px; height: 55px">

# Project 4: West Nile Virus Prediction

--- 
# Notebook 3

This notebook consists of:
- Cost benefit analysis of carring out regular spray excercises
- Conclusions and recommendations
---

# 1. Cost Benefit Analysis

## 1.1 Economical Impact of West Nile Virus

In the first 14 years since the virus was first detected in the USA in New York alone, hospitalised cases of WNV disease have cost close to \\$800 million in health care expenditures and lost productivity.<sup>1</sup>

## 1.2 Spray Costs

### 1.2.1 Pesticide Used
The pesticide used for the spray is **Zenifex E4**.<sup>2</sup><br>
It has a cost of **\\$0.92 / acre**.<sup>3</sup>

### 1.2.2 Spray Plan

For the sake of a 'worst case' calculation, we will assume that spray will be carried out at the locations all traps:
- 1km radius of spray for each trap
- Spray will be carried out from dusk till 1am.<sup>2</sup> In the summer months, dusk is around 8pm. Therefore the spray will be carried out over the course of approximately 5 hours.

In [2]:
# import the original train and test datasets
import pandas as pd
train = pd.read_csv('../assets/train.csv')
test = pd.read_csv('../assets/test.csv')

In [4]:
traps = pd.concat([train['Trap'], test['Trap']], axis=0)
print(traps.shape)
traps.head()

(126799,)


0    T002
1    T002
2    T007
3    T015
4    T015
Name: Trap, dtype: object

In [7]:
unique_traps = len(traps.unique())
print(f'There are {unique_traps} unique traps.')

There are 149 unique traps.


### 1.2.3 Labor Cost
- Typical pest control worker makes \\$19.65 / hour <sup>4</sup> - we will round this up to \\$20 / hour.
- One worker will handle one trap area of 1km radius.

### 1.2.4 Other Assumptions

- 1 acre = 0.00404686 km<sup>2</sup>
- Sprays will be conducted at each trap location 7 times a year, following 2013's schedule

### 1.2.5 Spray Cost Calculation

In [8]:
# Total spray area in square km
import math
total_spray_area_km = round(math.pi*unique_traps,2)
print(f'The total spray area is {total_spray_area_km} km\u00b2')

The total spray area is 468.1 km²


In [9]:
# Total spray area in acres
total_spray_area = round(total_spray_area_km/0.00404686,2)
print(f'The total spray area is {total_spray_area} acres')

The total spray area is 115669.93 acres


In [10]:
# zenivex cost for total spray area
pesticide_cost = round(0.92*total_spray_area,2)
print(f'The zenivex cost is ${pesticide_cost}.')

The zenivex cost is $106416.34.


In [11]:
# labor cost
labor = 5*20*136
print(f'The labor cost is ${labor}.')

The labor cost is $13600.


In [12]:
# one spray session cost
total_cost = labor + pesticide_cost
print(f'The total cost for 1 spray session is ${total_cost}.')

The total cost for 1 spray session is $120016.34.


In [13]:
# 7 sprays per year
annual_cost = round(total_cost*7,2)
print(f'The annual cost of spraying is ${annual_cost}.')

The annual cost of spraying is $840114.38.


## 1.3 Hospitalisation and Productivity Loss

### 1.3.1 Assumptions
- We will assume that not spraying will result 200 annual cases (225 cases in Chicago in 2002, when there were no spray efforts). <sup>5</sup>
- Of all WNV patients, only 1 in 150 cases will develop severe symptoms.<sup>6</sup> Therefore, we will assume all cases as non-severe. (conservative assumption). Non-severe cases incur a mean cost of \\$7,500 <sup>7</sup> in terms of medical bills and productivity loss.

### 1.3.2 Benefit Calculation

In [14]:
human_cost_annual = 7500*200
print(f'The annual cost if no spray is carried out is ${human_cost_annual}.')

The annual cost if no spray is carried out is $1500000.


In [15]:
savings = human_cost_annual - annual_cost
print(f'Spraying saves about ${savings} annually.')

Spraying saves about $659885.62 annually.


Even with a conservative estimate, the savings by conducting regular spray exercises will result in **$659,885.62** in annual savings.

# References
1. https://www.ajtmh.org/view/journals/tpmd/90/3/article-p402.xml
2. https://www.chicago.gov/city/en/depts/cdph/provdrs/healthy_communities/news/2020/august/city-to-spray-insecticide-thursday-to-kill-mosquitoes.html
3. https://www.centralmosquitocontrol.com/-/media/files/centralmosquitocontrol-na/us/resources-lit%20files/2015%20zenivex%20pricing%20brochure.pdf
4. https://www.indeed.com/career/pest-control-technician/salaries/Chicago--IL
5. https://idph.illinois.gov/envhealth/wnvsurveillance_humancases_02.htm
6. https://www.sciencedaily.com/releases/2014/02/140210184713.htm
7. https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3945683/table/T4/?report=objectonly

# 2. Conclusions and Recommendations

Using the given dataset, we analysed the number of WNV cases in Chicago, the clustering traps for mosquito, location of the pesticide spraying, the impact of pesticide spraying vs the WNV outbreak, location with high mosquito count and WNV cluster and effect of weather on the number of mosquito and WNV. Based on our findings, we discovered some useful insights about how the features contributed to the outbreak of West Nile Virus (WNV).

For instant, the feature 'month' gives a significant indicator in relationship with the number of WNV cases which indicated that the highest numbers of WNV present mostly appeared during the month of July, August and September. This could be due to July to September being the summer period for Chicago.

Based on the scores returned from Pycaret, the best model is Light Gradient Boosting Machine with highest AUC score 0.8260. The higher the AUC score, the better the model's performance at distinguishing between the positive and negative classes.

## 2.1 Further Research

We suggest to expand the study to other states and establish better insight on the number of mosquitos caught per trap, the life cycles of the mosquito and the weather pattern as to how, when and where to get a better predict target spraying model.

## 2.2 Better Adoption

Despite of spraying the pesticide, the best way to prevent WNV disease is to reduce the number of mosquitos around the home and to take personal precautions to avoid mosquito bites. Precautions include:
  * change water in vases/bowls on alternate days
  * turn over all water storage containers
  * remove water from flower pot plates on alternate days
  * cover bamboo pole holders when not in use.
  * clear blockages and put BTI insecticide in roof gutters monthly
  * spray insecticide in the dark corner around the house
  * apply insect repellent and wear covered clothes

## 2.3 Drone used in mosquito control

Recommend the use of drones in mosquito control with surveillance of nuisance mosquitos and potential vectors of pathogens. The insecticide application using drone to conduct surveillance and control in remote areas that are difficult to reach have been carried out by the GDG drone under the nuisance-reduction programme. The project was completed to nearly one thousand hectares of area producing mosquito larvae.

- [Source](https://sensorsandsystems.com/west-nile-virus-and-mosquito-nuisance-prevention-with-a-uav/?utm_source=rss&utm_medium=rss&utm_campaign=west-nile-virus-and-mosquito-nuisance-prevention-with-a-uav): West Nile Virus and Mosquito Nuisance Prevention with a UAV | Sensors and Systems
- [Source](https://www.terminix.com/blog/science-nature/drones-in-mosquito-control/): Mosquito Drone Technology | Terminix